reporter/internal/pdata/generate.go

```diff
@@ -69,28 +69,34 @@ func (p *Pdata) setProfile(
 ) {
 	// stringMap is a temporary helper that will build the StringTable.
 	// By specification, the first element should be empty.
+	strDicOffset := int32(dic.StringTable().Len())
 	stringMap := make(map[string]int32)
-	stringMap[""] = 0
+	if strDicOffset == 0 {
+		stringMap[""] = strDicOffset
+	}
 
 	// funcMap is a temporary helper that will build the Function array
 	// in profile and make sure information is deduplicated.
+	funcDicOffset := int32(dic.FunctionTable().Len())
 	funcMap := make(map[samples.FuncInfo]int32)
-	funcMap[samples.FuncInfo{Name: "", FileName: ""}] = 0
+	if funcDicOffset == 0 {
+		funcMap[samples.FuncInfo{Name: "", FileName: ""}] = 0
+	}
 
 	st := profile.SampleType().AppendEmpty()
 	switch origin {
 	case support.TraceOriginSampling:
-		st.SetTypeStrindex(getStringMapIndex(stringMap, "samples"))
-		st.SetUnitStrindex(getStringMapIndex(stringMap, "count"))
+		st.SetTypeStrindex(getStringMapIndex(stringMap, strDicOffset, "samples"))
```

Comment:
In Async-Profiler we created a [small abstraction](https://github.com/async-profiler/async-profiler/blob/master/src/index.h) to ease this process, and we fill the string table in order after all profiles have been processed. Maybe you could consider something similar? Passing around offsets seems a bit error prone.

Reply:
I consider this more of a hot patch and let the options open for https://github.com/open-telemetry/opentelemetry-ebpf-profiler/issues/515.

Reply:
It could be extended to any hashable type, including functions

Reply:
Yeah right. Every table in ProfilesDictionary could benefit from such an approach. Not just functions and locations.